src/main/java/uk/gov/pay/ledger/queue/EventMessageDto.java

```diff
@@ -21,19 +22,24 @@ public class EventMessageDto {
 
     private Boolean live;
 
+    @NotNull(message = "Field [timestamp] cannot be null")
     @JsonDeserialize(using = MicrosecondPrecisionDateTimeDeserializer.class)
     private ZonedDateTime timestamp;
 
+    @NotNull(message = "Field [resource_external_id] cannot be null")
     @JsonProperty("resource_external_id")
     private String externalId;
 
     @JsonProperty("parent_resource_external_id")
     private String parentExternalId;
 
+    @NotNull(message = "Field [event_type] cannot be null")
     public String eventType;
 
+    @NotNull(message = "Field [resource_type] cannot be null")
     public ResourceType resourceType;
 
+    @NotNull(message = "Field [event_details] cannot be null")
```

Comment:
`event_details` can be empty. Some events don't carry any payload. They just update the state of the resource.